### ISSUE TITLE
Indentation error with flow

### ISSUE THREAD

**Issue by ShaunLloyd** [yannickcr/eslint-plugin-react#1427]
### Versions:
`eslint`: 4.3.0
`eslint-plugin-react`: 7.1.0

### Problem

I'm using flow for my react native project and I'm coming across this weird error. One of my components has a lot of props that i'm destructuring onto their own individual lines. however, when I add the types to the destructured object I get an indentation error. However, fixing that error causes eslint to complain about the indentation of the jsx, which upon fixing that, tells me to put it back where it was.

#### Example:
```js
type Props = {
  email: string,
  password: string,
  error: string,
}

const SomeFormComponent = ({
  email,
  password,
  error,
}: Props) => (    // React/jsx-indent error found 0 spaces, expected 2
  // JSX
);
```

This only happens when the destructured props are on their own lines.  The example below causes no issues:

```js
type Props = {
  someProp: boolean,
}

const AnotherComponent = ({ someProp }: Props) => (
  // JSX
);
```
